bin/cli.js

```diff
@@ -118,8 +130,17 @@ async.waterfall([
 
       return documentedVersions;
     });
+    const latest = semver.getGreaterVersion(versions);
+    if (argv.target === 'version') {
+      console.log(latest);
+      return callback(STOP);
+    }
 
-    const gitReference = argv.configuration.getGitReferenceFromVersion(semver.getGreaterVersion(versions));
+    const gitReference = argv.configuration.getGitReferenceFromVersion(latest);
+    if (argv.target === 'reference') {
+      console.log(gitReference);
+      return callback(STOP);
```

Comment:
Sorry, I'm probably not being very clear here. Why not define the string code (if we're not using a number) at the top of the file:
```
const stopError = 'E_STOP`;
```
Then in the code just do:
```
return callback(throw new Error(stopError));
```
and then when testing, just:
```
if (!(error.message === stopError))
```
?

It cuts down the amount of code and doesn't leave you creating an object immediately which might never get used.

Reply:
this is how I had it initially, but I was not that happy with checking the message to determine the error type... `error.code` seems better for this case. I initialised the error on the top level to avoid having to set the code in every occasion you want to use stop. Maybe a compromise could be to use `.code` but only set it after constructing the error in the if blocks?

Reply:
I'm not sure I understand. `.code` is not a default property of a vanilla JS `Error`, whereas `.message` is, which doesn't require any extra code to set it as it's part of the constructor. If we were using a subclass inheriting from this (such as `typed-error`) because we needed to (like we do in a lot of backend components), I'd agree, but for a one-off to break the waterfall I'm not sure I see any benefit to it.

However, if you want to keep it like this, I'll approve.